src/game/moveFinding/findRookBaseMoves.js

```diff
@@ -1,3 +1,56 @@
-export default function findRookBaseMoves() {
-    return [];
+import BoardCoordinates from "../board/boardCoordinates";
+
+export default function findRookBaseMoves(boardSituation, startingCoordinates, color) {        
+        let currentBoard = boardSituation.position;
+        return checkMoves(currentBoard, startingCoordinates, color);
+}
+
+function checkMoves(currentBoard, startingCoordinates, color)
+{
+    let possibleMoves = []; //array with all possible moves
+    //call recursive function which checks possible moves in directions:left,right,up,down
+    makeStep(currentBoard, startingCoordinates.x, startingCoordinates.y, color, possibleMoves, "left");
+    makeStep(currentBoard, startingCoordinates.x, startingCoordinates.y, color, possibleMoves, "right");
+    makeStep(currentBoard, startingCoordinates.x, startingCoordinates.y, color, possibleMoves, "up");
+    makeStep(currentBoard, startingCoordinates.x, startingCoordinates.y, color, possibleMoves, "down");
+    //return array with all possible moves
+    return possibleMoves;
+}
+
+
+function makeStep(currentBoard, x, y, color, aggregator, direction){
+    //depending on direction increment or decrement proper variable
+    switch(direction)
+    {
+        case "up":
+            y++;
+            break;
+        case "down":
+            y--;
+            break;
+        case "left":
+            x--;
+            break;
+        case "right":
+            x++;
+            break;
+    }
+    //check if you are still on chessboard
+    if(x<0 || x>7 || y<0 || y>7)
+        return false;
+    //check for empty place on board
+    if (currentBoard[y][x]==null)
+    {
+        aggregator.push(new BoardCoordinates(x,y));
+        //you haven't found another chess piece - you can move further
+        return makeStep(currentBoard, x, y, color, aggregator, direction);
+    }
+    else
+    {
+        //check if you can capture this piece, if you can it is a legal move
+        if(currentBoard[y][x].color!=color)
+            aggregator.push(new BoardCoordinates(x,y));
+        return false;
```

Comment:
1. Zdarzają się sytuacje, gdy trudno uniknąć funkcji z kilkoma returnami, rozrzuconymi po różnych ifach.
To nie jest jedna z nich. Zasadniczo, ta funkcja w ogóle nie potrzebuje returna, co dopiero kilku (zauważ, że ta funkcja ZAWSZE zwraca false).
2. Zwracam Twoją uwagę na istnienie statycznej metody BoardCoordinates.valid(), która sprawdza, czy podany obiekt ma prawidłowe indeksy. Być może warto by jej użyć zamiast ręcznie sprawdzać granice dla x i y, zwłaszcza, że i tak korzystasz z klasy (a nawet tworzysz obiekty) BoardCoordinates
3. **Odwołując się do currentBoard odwróciłeś naturalną kolejność indeksów - wywołujesz [y][x] zamiast [x][y], w efekcie sprawdzasz nie te pola, które potem dodajesz.**

Reply:
Odnośnie punktu 3, czy nie jest tak że pierwszy indeks to indeks wiersza a drugi to indeks kolumny?

Reply:
W aktualnej wersji poprawiłem wszystko o czym wspomniałeś. Funkcje rekurencyjne nie są dla mnie do końca intuicyjne, ale jak już zwróciłeś mi na to uwagę, to rzeczywiście, żaden return nie był tam potrzebny. BoardCoordinates.valid() zwyczajnie mi umknęło, teraz w kodzie jest już ta funkcja zamiast niepotrzebnie dużego ifa.

Reply:
Nie. Dwuwymiarowa tablica to po prostu tablica tablic - wcale nie musi mieć przełożenia na wiersze i kolumny, a jeśli ma, to może być równie dobrze tablica wierszy co tablica kolumn. Wybiera się to, co ma więcej sensu w kontekście danych, które chce się reprezentować (co bardzo często istotnie oznacza, że pierwszy indeks odnosi się do wierszy - ale to po prostu kwestia tego, jak dużo operujemy danymi, o których myślimy jako o "wierszowej" strukturze, gdzie jest stała liczba kolumn do których dodaje się kolejne rekordy jako wiersze).

Reply:
indeksy też zmieniłem, więc wszystko powinno byc juz dobrze